llvm/docs/RISCVUsage.rst

```diff
@@ -431,3 +431,29 @@ line.  This currently applies to the following extensions:
 * ``Zvksg``
 * ``Zvksh``
 * ``Zvkt``
+
+Global Pointer (GP) Relaxation and the Small Data Limit
+=======================================================
+
+Some of the RISC-V psABI variants reserve ``gp`` (``x3``) for use as a "Global Pointer", to make generating data addresses more efficient.
+
+To use this functionality, you need to be doing all of the following:
+* Use the ``medlow`` (aka ``small``) code model;
```

Comment:
medany is supported with the bfd linker just not with lld.

Reply:
I'm keen we just document LLVM here (LLD and Clang).